From 1e1bfa93e805c3d7d7c9ffc8fdc2b907313a5152 Mon Sep 17 00:00:00 2001
From: Ben Sartor <ben@simlar.org>
Date: Wed, 13 May 2015 22:41:04 +0200
Subject: [PATCH 5/5] java exception handling in callbacks now logs backtrace

---
 coreapi/linphonecore_jni.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/coreapi/linphonecore_jni.cc b/coreapi/linphonecore_jni.cc
index 13244f5..62cfec0 100644
--- a/coreapi/linphonecore_jni.cc
+++ b/coreapi/linphonecore_jni.cc
@@ -1115,6 +1115,7 @@ private:
 	static inline void handle_possible_java_exception(JNIEnv *env, jobject listener)
 	{
 		if (env->ExceptionCheck()) {
+			env->ExceptionDescribe();
 			env->ExceptionClear();
 			ms_error("Listener %p raised an exception",listener);
 		}
-- 
2.1.4

